src/nodes/continuous_transition.jl

```diff
@@ -0,0 +1,105 @@
+
+Check CTMeta for more details on how to specify the transformation function that **must** return a matrix.
+
+```julia
+y ~ ContinuousTransition(x, a, W) where {meta = CTMeta(transformation, â)}
+```
+Interfaces:
+1. y - n-dimensional output of the ContinuousTransition node.
+2. x - m-dimensional input of the ContinuousTransition node.
+3. a - any-dimensional vector that casts into the matrix `A`. 
+4. W - `n×n`-dimensional precision matrix used to soften the transition and perform variational message passing.
+
+Note that you can set W to a fixed value or put a prior on it to control the amount of jitter.
+"""
+struct ContinuousTransition end
+
+const CTransition = ContinuousTransition
+
+@node ContinuousTransition Stochastic [y, x, a, W]
+
+@doc raw"""
+`CTMeta` is used as a metadata flag in `ContinuousTransition` to define the transformation function for constructing the matrix `A` from vector `a`.
+
+`CTMeta` requires a transformation function and the length of vector `a`, which acts as an expansion point for approximating the transformation linearly. If transformation appears to be linear, then no approximation is performed.
+
+Constructors:
+- `CTMeta(transformation::Function, â::Vector{<:Real})`: Constructs a `CTMeta` struct with the transformation function and allocated basis vectors.
+
+Fields:
+- `ds`: A tuple indicating the dimensionality of the ContinuousTransition (dy, dx).
+- `f`:  Represents the transformation function that transforms vector `a` into matrix `A`
+- `es`: A Vector of unit vectors used in the transformation process.
+
+The `CTMeta` struct plays a pivotal role in defining how the vector `a` is transformed into the matrix `A`, thus influencing the behavior of the `ContinuousTransition` node.
+"""
+struct CTMeta
+    ds::Tuple # dimensionality of ContinuousTransition (dy, dx)
+    f::Function # transformation function
+    es::Vector{<:AbstractVector} # unit vectors
+
+    # NOTE: this meta is not user-friendly, I don't like a vector
+    # perhaps making mutable struct with empty meta first will be better from user perspective
+    # meta for transformation of a vector to a matrix
+    function CTMeta(transformation::Function, â::Vector{<:Real})
+        dy, dx = size(transformation(â))
+        es = [StandardBasisVector(dy, i, one(eltype(first(â)))) for i in 1:dy]
+        return new((dy, dx), transformation, es)
+    end
+end
+
+getunits(meta::CTMeta)          = meta.es
+getdimensionality(meta::CTMeta) = meta.ds
+gettransformation(meta::CTMeta) = meta.f
+
+getjacobians(ctmeta::CTMeta, a) = process_Fs(gettransformation(ctmeta), a)
+process_Fs(f::Function, a) = [ForwardDiff.jacobian(a -> f(a)[i, :], a) for i in 1:size(f(a), 1)]
```

Comment:
That's something I've missed @bartvanerp. I was thinking about adding a flag to the constructor, i.e. precompute=true, that will precompute Jacobians which is useful in case of linear transformations.

Reply:
Should we include this to this PR though?

Reply:
Perhaps we will move it to a separate issue. See a comment below.